### PR TITLE
Gradle verify

### PR DESCRIPTION
**Pull request by hannesa2**
To increase trust

### CODE REVIEW

**Comment by hannesa2** [comment]
I see the action doesn't start, you have to enable it

image

**Comment by hannesa2** [comment]
Btw, I did it on my repo and see it has an issue !

https://github.com/hannesa2/ics-openvpn/pull/2

image


that's sad !
image

That's why I've to run ` ./gradlew wrapper --gradle-version=5.6.4` and commit the result as well, now it works on my fork

image

**Comment by schwabe** [comment]
Could you explain why this should be there. So far I used the gradlew that Google is providing and seeing that this is mainly a one man project and you need all other code I do anyway, so why checking the wrapper?

**Comment by hannesa2** [comment]
Here you have the full story https://github.com/gradle/wrapper-validation-action

In short: it fails if unknown Gradle Wrapper JAR files is found.

**Comment by hannesa2** [comment]
Yes, you know, but others don't.
Current master has an **unknown** Gradle Wrapper JAR ! There is nothing to explain.

**Comment by hannesa2** [comment]
Btw, why does Travis not start here ? I would expect a check during each pull request

**Comment by schwabe** [comment]
I will put this on ice for the time being:
 - I not accepting PR that change gradle binary stuff, so the risk for this project a single person repo is very small
  - I currently have no time to dive into the whole Github actions.

And also this PR does not just add the check but also modifies the gradle wrapper.

File: .github/workflows/gradle-wrapper-validation.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: "Validate Gradle Wrapper"
+on: [pull_request]
```
Comment (hannesa2) [inline .github/workflows/gradle-wrapper-validation.yml:2]: you have to change to

```
on:
  push:
    branches:
      - master
  pull_request:
    branches:
      - master
```

to see result of PR in overview. But I'll stop to waste my time here

**Comment by hannesa2** [comment]
>     * I not accepting PR that change gradle binary stuff, 
>      so the risk for this project a single person repo is very small

Your master has a modified jar file https://github.com/schwabe/ics-openvpn/pull/1196#issuecomment-629600673 
This is not a question of small or huge
 
> And also this PR does not just add the check but also modifies the gradle wrapper.

Because you modified the gradle jar file, it was necessary to run 
`./gradlew wrapper --gradle-version=5.6.4` and check-in result to have a successful build.

You see the comment [here](https://github.com/schwabe/ics-openvpn/pull/1196/commits/7f5e44ef67e7b6531fd0a6aac4d846eff77aef78). Btw, you could run it on your side, and observe what will be changed 

For all, who read this conversation:
Who uses a vpn client keeps an eye on security and should keep attention what's be use.

This repo has on `master` a modified jar file and the  **maintainer doesn't want to change this**
# Attention, this is a serious security issue !

**Comment by schwabe** [comment]
@hannesa2 If that is the serious issue why did you point that out in the initial commit? If this fixes a serious issue then it would be better to write what the issue is instead of writing "Increase trust". 

And if you look at the git log you will see that the file has never been modified after its initial commit of 2014.  

And if you read the link you posted you will find the answer why this is "modified": 

> Note: When initially applying this action to your project, if your gradle-wrapper.jar was generated by Gradle 3.3 to 4.0, the check will fail. This is because these gradle-wrapper.jar versions were dynamically generated by Gradle in a non-reproducible manner. 

So please don't make a bigger deal out of it than it is. I just don't have time right to deal with this.

**Comment by hannesa2** [comment]
> So please don't make a bigger deal out of it than it is

Sorry it's a big deal on a security app

**Comment by schwabe** [comment]
if you really think that a project having an old file that came from a time when gradle did not have reprocidible builds is a big deal, go ahead, request a CVE and report that as vulnerability. I am not spending time on this.

**Pull request closed without merging**